Tests/Provider/AbstractProviderTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Tests\Provider;
+
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Sonata\MediaBundle\Tests\Form\Type\AssertFormReferenceCompatibleWithSymfony28Trait;
+use Symfony\Component\Form\FormBuilder;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * Class AbstractTypeTest.
+ *
+ * @author Virgile Vivier <virgilevivier@gmail.com>
+ */
+abstract class AbstractProviderTest extends \PHPUnit_Framework_TestCase
+{
+    use AssertFormReferenceCompatibleWithSymfony28Trait;
+
+    /**
+     * @var FormBuilder
+     */
+    protected $formBuilder;
+
+    /**
+     * @var FormMapper
+     */
+    protected $formMapper;
+
+    /**
+     * @var FormTypeInterface
+     */
+    protected $formType;
+
+    /**
+     * @var MediaProviderInterface
+     */
+    protected $provider;
+
+    public function setUp()
+    {
+        $this->formMapper = $this->getMockBuilder('Sonata\AdminBundle\Form\FormMapper')->disableOriginalConstructor()->getMock();
+        $this->formMapper
+            ->expects($this->any())
+            ->method('add')
+            ->will($this->returnCallback(function ($name, $type = null) {
+                if (null !== $type) {
+                    $this->assertFormReferenceCompatibleWithSymfony28($type);
+                }
+            }));
+
+        $this->formBuilder = $this->getMockBuilder('Symfony\Component\Form\FormBuilder')->disableOriginalConstructor()->getMock();
+        $this->formBuilder
+            ->expects($this->any())
+            ->method('add')
+            ->will($this->returnCallback(function ($name, $type = null) {
+                if (null !== $type) {
+                    $this->assertFormReferenceCompatibleWithSymfony28($type);
+                }
+            }));
+
+
+        // Second test
+        $this->formBuilder->expects($this->any())->method('getOption')->willReturn('api');
+
+        $this->provider = $this->getProvider();
+    }
+
+    abstract public function getProvider();
+
+    public function testBuildEditFormUseFormReferenceCompatibleWithSymfony28()
```

Comment:
> Don't make this public! It's a BC hack and should at most be private.

@core23 I think it should be public because it's a test case and PHPUnit have to execute it.